src/Sylius/Bundle/ApiBundle/spec/EventSubscriber/ProductDeletionEventSubscriberSpec.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\EventSubscriber;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Exception\ProductCannotBeRemoved;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Promotion\Checker\ProductPromotionRuleCheckerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+final class ProductDeletionEventSubscriberSpec extends ObjectBehavior
+{
+    function let(ProductPromotionRuleCheckerInterface $productPromotionRuleChecker): void
+    {
+        $this->beConstructedWith($productPromotionRuleChecker);
+    }
+
+    function it_does_not_throw_exception_when_product_is_not_being_deleted(
+        ProductInterface $product,
+        Request $request,
+        ProductPromotionRuleCheckerInterface $productPromotionRuleChecker,
+        HttpKernelInterface $kernel,
+    ): void {
+        $request->getMethod()->willReturn(Request::METHOD_POST);
+
+        $event = new ViewEvent(
+            $kernel->getWrappedObject(),
+            $request->getWrappedObject(),
+            HttpKernelInterface::MASTER_REQUEST,
```

Comment:
We should use `HttpKernelInterface::MAIN_REQUEST` as `MASTER_REQUEST` is deprecated